### PR TITLE
Add quick start example using Vagrant

### PR DESCRIPTION
**Pull request by twpayne**
Signed-off-by: Tom Payne <tom@isovalent.com>

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Vagrant.configure("2") do |config|
+    config.vm.box = "ubuntu/impish64"
```
Comment (aditighag) [inline Vagrantfile:2]: What kernel version does this box come with?

File: main.go [hunk 2]
```diff
@@ -25,8 +25,8 @@ import (
 	"github.com/cilium/pwru/internal/pwru"
 )
 
-//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang KProbePWRU ./bpf/kprobe_pwru.c -- -DOUTPUT_SKB -I./bpf/headers
-//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang KProbePWRUWithoutOutputSKB ./bpf/kprobe_pwru.c -- -I./bpf/headers
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang-12 KProbePWRU ./bpf/kprobe_pwru.c -- -DOUTPUT_SKB -I./bpf/headers
```
Comment (aditighag) [inline main.go:28]: I don't think we want to pin this clang binary. Can we add a symlink in the Vagrant example instead?

Comment (twpayne) [reply main.go]: Done.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    config.vm.define :impish64
+    config.vm.hostname = "impish64"
+    config.vm.synced_folder ".", "/pwru"
+    config.vm.provision "shell", inline: <<-SHELL
+      export DEBIAN_FRONTEND=noninteractive
+      apt-get update
+      apt-get install -y clang-12 golang
+      ln -sf clang-12 /usr/bin/clang
```
[the remark this answers is not in the document]
Comment (twpayne) [reply Vagrantfile]: Thanks! That's much cleaner. Done.

File: README.md [hunk 4]
```diff
@@ -11,6 +11,62 @@ after installing an IP tables rule:
...
+
+2. Connect to the Vagrant box:
+   ```console
+   $ vagrant ssh
+   ```
+
+3. Build `pwru`:
+   ```console
+   $ cd /pwru
+   $ go generate
```
[the remark this answers is not in the document]
Comment (twpayne) [reply README.md:33]: Yes, it's installed at line 9 of the `Vagrantfile`.

**Pull request merged**